shell/server/installer.js

```diff
@@ -70,28 +84,40 @@ deletePackage = function (packageId) {
   }
 }
 
-startInstall = function (packageId, url, fromBeginning, appId) {
-  if (packageId in installers) {
-    return;
+startInstall = function (packageId, url, retryFailed) {
+  // Mark package for possible installation.
+
+  var fields = {status: "download", progress: 0, url: url};
+
+  if (retryFailed) {
+    Packages.update({_id: packageId, status: "failed"}, {$set: fields});
+  } else {
+    fields._id = packageId;
+    Packages.insert(fields);
```

Comment:
wrapped in a try/catch that logs server-side. I think on the client side the user will just see the install appear to start normally, so this should work...